src/lava/magma/core/model/py/ports.py

```diff
@@ -208,9 +251,102 @@ def write(self, data: int, idx: int):
-    SCALAR_DENSE = PyRefPortScalarDense
-    SCALAR_SPARSE = PyRefPortScalarSparse
+class PyVarPort(AbstractPyPort):
+    """Python implementation of InPort used within AbstractPyProcessModel.
+    If buffer is empty, recv() will be blocking.
+    """
+
+    def __init__(self,
+                 var_name: str,
+                 csp_send_port: ty.Optional[CspSendPort],
+                 csp_recv_port: ty.Optional[CspRecvPort], *args):
+        self._csp_recv_port = csp_recv_port
+        self._csp_send_port = csp_send_port
+        self.var_name = var_name
+        super().__init__(*args)
+
+    VEC_DENSE: ty.Type["PyVarPortVectorDense"] = None
+    VEC_SPARSE: ty.Type["PyVarPortVectorSparse"] = None
+    SCALAR_DENSE: ty.Type["PyVarPortScalarDense"] = None
+    SCALAR_SPARSE: ty.Type["PyVarPortScalarSparse"] = None
+
+    def start(self):
+        if self._csp_send_port is not None:
+            self._csp_send_port.start()
+        if self._csp_recv_port is not None:
+            self._csp_recv_port.start()
+
+    def join(self):
+        if self._csp_send_port is not None:
+            self._csp_send_port.join()
+        if self._csp_recv_port is not None:
+            self._csp_recv_port.join()
+
+    def service(self):
+        pass
+
+
+class PyVarPortVectorDense(PyVarPort):
+    def service(self):
+        """Sets the received value to the given var or sends the value of the
+        var to the csp_send_port, depending on the received header information
+        of the csp_recv_port."""
+
+        # Inspect incoming data
+        if self._csp_send_port is not None and self._csp_recv_port is not None:
+            while self._csp_recv_port.probe():
+                cmd = enum_to_np(self._csp_recv_port.recv()[0])
+
+                # Set the value of the Var with the given data
+                if np.array_equal(cmd, VarPortCmd.SET):
+                    data = self._csp_recv_port.recv()
+                    setattr(self._process_model, self.var_name, data)
+                elif np.array_equal(cmd, VarPortCmd.GET):
+                    data = getattr(self._process_model, self.var_name)
+                    self._csp_send_port.send(data)
+                else:
+                    raise ValueError(f"Wrong Command Info Received : {cmd}")
+
+
+class PyVarPortVectorSparse(PyVarPort):
+    def recv(self) -> ty.Tuple[np.ndarray, np.ndarray]:
+        pass
+
+    def peek(self) -> ty.Tuple[np.ndarray, np.ndarray]:
+        pass
+
+
+class PyVarPortScalarDense(PyVarPort):
+    def recv(self) -> int:
+        pass
+
+    def peek(self) -> int:
+        pass
+
+
+class PyVarPortScalarSparse(PyVarPort):
+    def recv(self) -> ty.Tuple[int, int]:
+        pass
+
+    def peek(self) -> ty.Tuple[int, int]:
+        pass
+
+
+PyVarPort.VEC_DENSE = PyVarPortVectorDense
+PyVarPort.VEC_SPARSE = PyVarPortVectorSparse
+PyVarPort.SCALAR_DENSE = PyVarPortScalarDense
+PyVarPort.SCALAR_SPARSE = PyVarPortScalarSparse
+
+
+class RefVarTypeMapping:
```

Comment:
Yes, but I preferred a class, since we also store the mapping and I prefer to import classes over functions.

Unless you prefer to include the function in the PyRefPort or PyVarPort class.

Reply:
Well... perhaps more efficient to build the dictionary only once in a class.